modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeFactory.java

```diff
@@ -163,30 +161,72 @@ public Type getJavaClass(RelDataType type) {
     }
 
     /**
-     * Gets ColumnType type for given class.
+     * Gets ColumnType type for RelDataType.
      *
      * @param relType Rel type.
      * @return ColumnType type or null.
      */
-    public ColumnType columnType(RelDataType relType) {
+    public static ColumnType relDataTypeToColumnType(RelDataType relType) {
         assert relType != null;
+        assert relType instanceof BasicSqlType
+                || relType instanceof IntervalSqlType : "Not supported yet."; // Implement Class->ColumnType mapping if failed.
 
-        Type javaType = getResultClass(relType);
-
-        if (javaType == byte[].class) {
-            return relType.getPrecision() == PRECISION_NOT_SPECIFIED ? ColumnType.blobOf() :
-                ColumnType.blobOf(relType.getPrecision());
-        } else if (javaType == String.class) {
-            return relType.getPrecision() == PRECISION_NOT_SPECIFIED ? ColumnType.string() :
-                ColumnType.stringOf(relType.getPrecision());
-        } else if (javaType == BigInteger.class) {
-            return relType.getPrecision() == PRECISION_NOT_SPECIFIED ? ColumnType.numberOf() :
-                ColumnType.numberOf(relType.getPrecision());
-        } else if (javaType == BigDecimal.class) {
-            return relType.getPrecision() == PRECISION_NOT_SPECIFIED ? ColumnType.decimalOf() :
-                ColumnType.decimalOf(relType.getPrecision(), relType.getScale());
-        } else {
-            return SchemaConfigurationConverter.columnType((Class<?>) javaType);
+        switch (relType.getSqlTypeName()) {
+            case BOOLEAN:
+                throw new IllegalArgumentException("Type is not supported yet.");
+            case TINYINT:
+                return ColumnType.INT8;
+            case SMALLINT:
+                return ColumnType.INT16;
+            case INTEGER:
+                return ColumnType.INT32;
+            case BIGINT:
+                return ColumnType.INT64;
+            case DECIMAL:
+                return relType.getPrecision() == PRECISION_NOT_SPECIFIED ? ColumnType.decimalOf() :
```

Comment:
I believe `PRECISION_NOT_SPECIFIED` should be valid precision for varying types only, like `VARCHAR` or `VARBINARY`. Probably we could put an assertion here instead, WDYT?